### PR TITLE
Feature/dotnet6 upgrade

### PR DESCRIPTION
**Pull request by theczorn**
Taking @stephenweinrich 's changes and PR.

After speaking with Karlin Kappala, there isn't a way for us to keep using Specflow. Specflow needs to update their libraries to support 6.x.

PR should be shoved into the `main` branch as the golden copy. I've spun off a fork of `main-5.x` for posterity.

### CODE REVIEW

File: src/BlackSlope.Api.Common/BlackSlope.Api.Common.csproj [hunk 1]
```diff
@@ -1,8 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
-    <LangVersion>9.0</LangVersion>
```
Comment (theczorn) [inline src/BlackSlope.Api.Common/BlackSlope.Api.Common.csproj:5]: No reason for us to keep targeting 9.x. We ought default to the latest language impl unless we have a real good reason.

File: src/BlackSlope.Api.Common/Extensions/BlackSlopeServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -43,14 +43,13 @@ public static class BlackSlopeServiceCollectionExtensions
             services.AddMvc()
             .AddJsonOptions(options =>
             {
-                options.JsonSerializerOptions.IgnoreNullValues = true;
+                options.JsonSerializerOptions.DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull;
```
Comment (theczorn) [inline src/BlackSlope.Api.Common/Extensions/BlackSlopeServiceCollectionExtensions.cs:46]: Serializer options impl has changed to this.

File: src/BlackSlope.Api.Common/Extensions/BlackSlopeServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -43,14 +43,13 @@ public static class BlackSlopeServiceCollectionExtensions
...
             {
-                options.JsonSerializerOptions.IgnoreNullValues = true;
+                options.JsonSerializerOptions.DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull;
                 options.JsonSerializerOptions.PropertyNameCaseInsensitive = true;
                 options.JsonSerializerOptions.Converters
                     .Add(new JsonStringEnumConverter());
                 options.JsonSerializerOptions.Converters
                     .Add(new VersionJsonConverter());
-            })
-            .SetCompatibilityVersion(CompatibilityVersion.Version_3_0);
```
Comment (theczorn) [inline src/BlackSlope.Api.Common/Extensions/BlackSlopeServiceCollectionExtensions.cs:53]: 5.x offered compatibility with 3.x. 6.x doesn't offer this, though I'd assume 7.x would.
Scrubbed.

File: src/BlackSlope.Api.Tests/OperationsTests/MoviesTests/ValidatorsTests/CreateMovieViewModelValidatorShould.cs [hunk 4]
```diff
@@ -26,17 +26,19 @@ public CreateMovieViewModelValidatorShould()
         public void Fail_when_title_is_null()
         {
             _modelViewModel.Title = null;
-            var failures = _movieViewModelValidator
-                .ShouldHaveValidationErrorFor(x => x.Title, _modelViewModel)
+            var result = _movieViewModelValidator.TestValidate(_modelViewModel);
+            var failures = result
+                .ShouldHaveValidationErrorFor(x => x.Title)
```
Comment (theczorn) [inline src/BlackSlope.Api.Tests/OperationsTests/MoviesTests/ValidatorsTests/CreateMovieViewModelValidatorShould.cs:31]: FluentValidation test flow has changed to this syntax.

**Comment by rick-brown-slalom** [comment]
You'll need to update the docker images as well to 6.0

**Pull request merged**